tests/test_scripts.py

```diff
@@ -99,6 +100,68 @@ def test_parse_command_line_no_identifiers(self):
         eq_(Identifier.OVERDRIVE_ID, parsed.identifier_type)
 
 
+class TestPatronInputScript(DatabaseTest):
+
+    def test_parse_patron_list(self):
+        """Test that patrons can be identified with any unique identifier."""
+        p1 = self._patron()
+        p1.authorization_identifier = self._str
+        p2 = self._patron()
+        p2.username = self._str
+        p3 = self._patron()
+        p3.external_identifier = self._str
+        args = [p1.authorization_identifier, 'no-such-patron',
+                '', p2.username, p3.external_identifier]
+        patrons = PatronInputScript.parse_patron_list(
+            self._db, args
+        )
+        eq_([p1, p2, p3], patrons)
+
+        eq_([], PatronInputScript.parse_patron_list(self._db, []))
+
+    def test_parse_command_line(self):
+        p1 = self._patron()
+        p2 = self._patron()
+        p1.authorization_identifier = self._str
+        p2.authorization_identifier = self._str
+        # We pass in one patron identifier on the command line...
+        cmd_args = [p1.authorization_identifier]
+        # ...and another one into standard input.
+        stdin = MockStdin(p2.authorization_identifier)
+        parsed = PatronInputScript.parse_command_line(
+            self._db, cmd_args, stdin
+        )
+        eq_([p1, p2], parsed.patrons)
+
+    def test_parse_command_line_no_identifiers(self):
+        parsed = PatronInputScript.parse_command_line(
+            self._db, [], MockStdin()
+        )
+        eq_([], parsed.patrons)
+
+
+    def test_do_run(self):
+        """Test that PatronInputScript.do_run() calls process_patron()
+        for every patron designated by the command-line arguments.
+        """
+        class MockPatronInputScript(PatronInputScript):
+            def process_patron(self, patron):
+                patron.processed = True
+        p1 = self._patron()
+        p2 = self._patron()
+        p3 = self._patron()
+        p3.processed = False
+        p1.authorization_identifier = self._str
+        p2.authorization_identifier = self._str
+        cmd_args = [p1.authorization_identifier, p2.authorization_identifier]
+        stdin = MockStdin(p2.authorization_identifier)
```

Comment:
Did you mean to have p2 in both cmd_args and stdin? Might be good to test one that's only in stdin.